### PR TITLE
Improve property resolution for file fetching

### PR DESCRIPTION
**Pull request by mpolgar**
- Use regex to only fetch database properties that can potentially match the context being resolved when rendering a file. This gives a significant speed-up for properties with many contexts/values, and is negligible otherwise.
- Consolidate file rendering logic between the /rawFile and /getFilePreview endpoints (previous improvements did not apply to /getFilePreview).
- Make non-resolution of properties more explicit (in line with non-resolution of embedded file properties).

### CODE REVIEW

**Review by hakgagik — APPROVED** [review]

File: core/src/main/java/com/confighub/core/resolver/Context.java [hunk 1]
```diff
@@ -135,6 +140,32 @@ protected boolean containsLevelAtDepth(Depth d, String levelName)
         return false;
     }
 
+    protected String getRegexPattern() {
```
Comment (wilstoff) [inline core/src/main/java/com/confighub/core/resolver/Context.java:143]: is this only for the db pattern, may want to specify a more specific name

File: core/src/main/java/com/confighub/core/utils/FileUtils.java [hunk 2]
```diff
@@ -200,33 +161,32 @@ private static String previewFile(final Context context,
...
                     if (property.isEncrypted())
                     {
                         String spName = property.getPropertyKey().getSecurityProfile().getName();
-                        if (null != passwords && passwords.containsKey(spName))
-                        {
-                            String password = passwords.get(spName).get(0);
-                            if (!Utils.isBlank(password))
-                                property.decryptValue(password);
-                        }
+                        if (passwords.containsKey(spName))
```
Comment (wilstoff) [inline core/src/main/java/com/confighub/core/utils/FileUtils.java:171]: what's the goal here?  seems like we are adjusting this slightly for blank passwords

Comment (mpolgar) [reply core/src/main/java/com/confighub/core/utils/FileUtils.java:171]: So the passwords map will never be null, based on how this function is called. This function combines the functionality of the old resolveFile()/previewFile() functions. In resolveFile(), there's no extra check for if the password is blank.

File: core/src/main/java/com/confighub/core/resolver/Context.java (hunk 1 above)
Comment (mpolgar) [reply core/src/main/java/com/confighub/core/resolver/Context.java]: Changed in latest version

**Review by wilstoff — APPROVED** [review]

**Review by belvedere-trading-user — APPROVED** [review]

**Pull request merged**